### PR TITLE
[Fanout] Update Mellanox OYNX fanout deploy code and templates

### PR DESCRIPTION
**Pull request by keboliu**
### Description of PR

Summary:
Fixes # (issue)
Add Mellanox ONYX fanout deploy code and update the existing templates

### Type of change

- [] Bug fix
- [] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

#### How did you do it?
Add a fanout_mlnx.yml for Mellanox ONYX fanout switch deploy
Add a fanout configuration templates
```
ansible/roles/fanout/templates/mlnx_check_pfcwd_fanout.j2
ansible/roles/fanout/templates/mlnx_deploy_pfcwd_fanout.j2
ansible/roles/fanout/templates/mlnx_fanout.j2
```
update existing PFC watch dog yml

#### How did you verify/test it?
run command on fanout swith
`ansible-playbook -i lab fanout.yml -l <switch_name> -e os=mellanox --become --tags deploy -vvvv`

### CODE REVIEW

File: ansible/roles/fanout/tasks/mlnx/deploy_pfcwd_fanout.yml [hunk 1]
```diff
@@ -35,16 +33,14 @@
...
 
-    - name: Update storage driver for Docker
-      command: 'sed -i s/\"storage-driver\":\ \"vfs\",/\"storage-driver\":\ \"devicemapper\",\\n\ \ \ \ \"storage-opts\":\ [\\n\ \ \ \ \ \ \ \ \"dm.fs=ext4\"\\n\ \ \ \ ],\/g /opt/tms/bin/docker_config.json'
+- name: Update storage driver for Docker
+  command: 'sed -i s/\"storage-driver\":\ \"vfs\",/\"storage-driver\":\ \"devicemapper\",\\n\ \ \ \ \"storage-opts\":\ [\\n\ \ \ \ \ \ \ \ \"dm.fs=ext4\"\\n\ \ \ \ ],\/g /opt/tms/bin/docker_config.json'
 
-  always:
-    - name: Remount FS back to read-only
-      command: mount -r -o remount /
+- name: Remount FS back to ro
```
Comment (stepanblyschak) [inline ansible/roles/fanout/tasks/mlnx/deploy_pfcwd_fanout.yml]: Why removed always?

Comment (keboliu) [reply ansible/roles/fanout/tasks/mlnx/deploy_pfcwd_fanout.yml]: I merged from the file "ansible/roles/fanout/tasks/mlnx_deploy_pfcwd_fanout.yml"  of the local repo,  I am assuming what we are using in the local repo should be the latest, is this the case here?

Comment (stepanblyschak) [reply ansible/roles/fanout/tasks/mlnx/deploy_pfcwd_fanout.yml]: @keboliu no, you are looking at ansible/roles/fanout/tasks/mlnx_deploy_pfcwd_fanout.yml, this is outdated, ansible/roles/fanout/tasks/mlnx/deploy_pfcwd_fanout.yml is correct and used file.
You can remove ansible/roles/fanout/tasks/mlnx_deploy_pfcwd_fanout.yml from local repository

Comment (keboliu) [reply ansible/roles/fanout/tasks/mlnx/deploy_pfcwd_fanout.yml]: @stepanblyschak you are correct, I thought we are using mlnx_pfcwd*.yml, I reverted the change on that two files.

**Comment by stepanblyschak** [comment]
@keboliu Small comment - 'OYNX' -> 'ONYX' in PR title :)

**Comment by keboliu** [comment]
> @keboliu Small comment - 'OYNX' -> 'ONYX' in PR title :)

sure, thanks.

**Pull request closed without merging**